main.go

```diff
@@ -1,83 +1,47 @@
 package main
 
 import (
-	"bufio"
 	"fmt"
-	"os"
+	"reflect"
 )
 
-// Data structure = linked list as a stack [data+head]->[2]->[1]->[0]
-
-// acts as head of list & contains DB
-type HeadDB struct {
-	// access to stack of operations
-	current *Node
-	// most recent ops for same key
-	data map[string]string
-}
-
 // represents transaction with a hash of operations
 type Node struct {
 	parent *Node
-	ops    map[string]struct
-}
-
-type Read struct {
-}
-
-type Write struct {
-	value string
-}
-
-type Delete struct {
+	data   map[string]string
 }
 
-// might need an interface for operations (?)
-
-func (h *HeadDB) WriteTo(k,v string) *HeadDB {
-	// reference new Write Struct
-	writeOp := Write{value: v}
-	// access current ops list
-	list = &h.current.ops
-	*list[k] = writeOp
+type Root struct {
```

Comment:
what function does this struct serve? Would it be easier to just maintain a pointer to the current active transaction?